internal/server/middleware_has_country_permission.go

```diff
@@ -0,0 +1,54 @@
+package server
+
+import (
+	"net/http"
+
+	"github.com/nrc-no/notcore/internal/logging"
+	"github.com/nrc-no/notcore/internal/utils"
+	"go.uber.org/zap"
+)
+
+func hasCountryPermissionMiddleware() func(handler http.Handler) http.Handler {
```

Comment:
We should probably pass the required permission here 
```suggestion
func hasCountryPermissionMiddleware(p auth.Permission) func(handler http.Handler) http.Handler {
```

Reply:
I don't think this works because of how the handlers are setup. We have a single handler (therefore shared middleware) for each route. The same handler is used for all http methods, which is why we check the http method in the middleware. We could split out the router:

```
individualsRouter.Path("/{individual_id}").Methods("GET").Handler(withMiddleware(
	handlers.HandleIndividual(tpl, individualRepo),
	ensureSelectedCountryMiddleware(),
	hasCountryPermissionMiddleware("read"),
))
individualsRouter.Path("/{individual_id}").Methods("PUT").Handler(withMiddleware(
	handlers.HandleIndividual(tpl, individualRepo),
	ensureSelectedCountryMiddleware(),
	hasCountryPermissionMiddleware("write"),
))
```

Reply:
If we do this we could also split out the handlers so that they each have a reduced responsibility (one handler per route/method pair)